code/datums/research_upgrade_datum.dm

```diff
@@ -0,0 +1,174 @@
+/datum/research_upgrades
+	///unique to every upgrade. not the name of the item. name of the upgrade
+	var/name = "Upgrade."
+	///name of upgrades, not items. Items are at research_upgrades.dm somewhere in item folder.
+	var/desc = "something is broken. yippee!!"
+	///which behavior should this type follow. Should this be completely excluded from the buy menu? should it be one of the dropdown options? This is also what gets passed to the initizialize of an item, this can be any number *but* it cannot be -1 or -2, as it messes with the buy menu.
+	var/behavior = RESEARCH_UPGRADE_EXCLUDE_BUY // should this be on the list?
+	/// the price of the upgrade, refer to this: 500 is a runner, 8k is queen. T3 is usually 3k, woyer is 2k.
+	var/value_upgrade = 1000
+	/// actual path to the item.(upgrade)
+	var/item_reference
+	///In which tab the upgrade should be.
+	var/upgrade_type
+	///Clearance requirment to buy this upgrade. 5x is level 6. Why is it not that way? no one knows.
+	var/clearance_req = 5
+	///The change of price for item per purchase, recommended for mass producing stuff or limited upgrade.
+	var/change_purchase = 0
+	///the minimum price which we cant go any cheaper usually dont need to set this if change price is 0 or positive
+	var/minimum_price = 0
+	///the maximum price which we cant go any more expensive, usually dont need to set this if change price is 0 or negative
+	var/maximum_price = INFINITY
+
+/datum/research_upgrades/machinery
+	name = "Machinery"
+	behavior = RESEARCH_UPGRADE_CATEGORY // one on the dropdown choices you get
+
+/datum/research_upgrades/machinery/autodoc
+	name = "AutoDoc Upgrade"
+	behavior = RESEARCH_UPGRADE_EXCLUDE_BUY
+	item_reference = /obj/item/research_upgrades/autodoc
+	upgrade_type = ITEM_MACHINERY_UPGRADE
+
+/datum/research_upgrades/machinery/autodoc/internal_bleed
+	name = "AutoDoc Internal Bleeding Repair"
+	desc = "A data and instruction set for the Autodoc, making it capable of rapidly fixing internal bleeding."
+	behavior = RESEARCH_UPGRADE_TIER_1
+	value_upgrade = 200
+	clearance_req = 1
+
+/datum/research_upgrades/machinery/autodoc/broken_bone
+	name = "AutoDoc Bone Fracture Repair"
+	desc = "A data instruction set for the Autodoc, making it capable of setting fractures and applying bonegel."
+	behavior = RESEARCH_UPGRADE_TIER_2
+	value_upgrade = 2000
+	clearance_req = 3
+
+/datum/research_upgrades/machinery/autodoc/organ_damage
+	name = "AutoDoc Broken Organ Repair"
+	desc = "A data and instruction set for the Autodoc, making it capable of fixing organ damage."
```

Comment:
AutoDoc or Autodoc, pick one!!

Reply:
death to the Autodoc, glory to the new AutoDoc!